yarn-project/aztec-nr/aztec/src/history/note_inclusion.nr

```diff
@@ -10,31 +10,42 @@ use crate::{
     oracle::get_membership_witness::get_note_hash_membership_witness,
 };
 
-pub fn prove_note_commitment_inclusion(
-    note_commitment: Field,
-    block_number: u32, // The block at which we'll prove that the note exists
+pub fn prove_note_inclusion<Note, N>(
```

Comment:
Step 2 and 3 of these are the same between the functions and we are just changing the header so we could do a shared logic function for them.

```rust
fn _note_inclusion(
  note: Note,
  header: Header
) where Note: NoteInterface {
    // 1) Compute note_hash
    let note_hash = compute_note_hash_for_consumption(note);

    // 2) Get the membership witness of the note in the note hash tree
    let witness = get_note_hash_membership_witness(
        header.global_variables.block_number as u32,
        note_hash
    );

    // 3) Prove that the commitment is in the note hash tree
    assert_eq(
        header.state.partial.note_hash_tree.root,
        compute_merkle_root(note_hash, witness.index, witness.path), 
        "Proving note inclusion failed"
    );
}
```

Then the difference between the two is just how you fetch the initial header. 

```rust 
pub fn prove_note_inclusion_at<Note, N>(
    note: Note,
    block_number: u32, // The block at which we'll prove that the note exists
    context: PrivateContext
) where Note: NoteInterface {
   _note_inclusion(note, context.get_header_at(block_number));
}

pub fn prove_note_inclusion<Note, N>(
    note: Note,
    context: PrivateContext
) where Note: NoteInterface {
   _note_inclusion(note, context.get_header());
}
```
```